guides/common/modules/proc_creating-a-playbook-with-modules-from-project-ansible-collections.adoc

```diff
@@ -0,0 +1,124 @@
+.. Create an Ansible vault.
+For example, to create a vault named _My_Vault.yml_:
++
+[subs="+quotes,attributes"]
+----
+$ ansible-vault create _My_Vault.yml_
+----
+.. After the `ansible-vault create` command opens an editor, provide the required parameters in the _key_: _value_ format.
++
+If you want to authenticate with {Project} user credentials:
++
+[source,ini,subs="+quotes,attributes"]
+----
+_My_Server_URL_: _https://{foreman-example-com}_
+_My_Username_: _My_Admin_User_Account_
+_My_Password_: _My_Admin_Password_
+----
++
+If you want to authenticate with a Kerberos ticket or a Personal Access Token:
++
+[source,ini,subs="+quotes,attributes"]
+----
+_My_Server_URL_: _https://{foreman-example-com}_
+----
+.. Save the changes, and close the editor.
+Ansible encrypts the data in the vault.
+. Create a playbook file that references the vault file:
++
+[NOTE]
+====
+The following YAML snippets include the `module_defaults` keyword to define vault parameters for all modules from the {ansible-module-defaults-group} group that are used in the playbook.
+Module defaults groups simplify parameter management by enabling you define common parameters to groups of modules rather than having to pass the parameters to each module individually.
+====
+.. Provide details on how to authenticate to the {Project} API.
++
+If you are authenticating with {Project} user credentials, map the `username`, `password`, and `server_url` parameters to the contents of _My_Vault.yml_:
++
+[source,yaml,subs="+quotes,attributes"]
+----
+- name: _My Playbook_
+  hosts: _localhost_
+  vars_files:
+    - _My_Vault.yml_
+  module_defaults:
+    group/{ansible-module-defaults-group}:
+      username: "{{ _My_Username_ }}"
+      password: "{{ _My_Password_ }}"
+      server_url: "{{ _My_Server_URL_ }}"
+  tasks:
+----
++
+If you are authenticating with a Kerberos ticket, map the `server_url` parameter to the contents of _My_Vault.yml_ and use `use_gssapi: true` to enable Kerberos authentication:
++
+[source,yaml,subs="+quotes,attributes"]
+----
+- name: _My Playbook_
+  hosts: _localhost_
+  vars_files:
+    - _My_Vault.yml_
+  module_defaults:
+    group/{ansible-module-defaults-group}:
+      server_url: "{{ _My_Server_URL_ }}"
+      use_gssapi: true
+  tasks:
+----
++
+If you are authenticating with a {Project} Personal Access Token, map the `username` parameter to the contents of _My_Vault.yml_:
++
+[source,yaml,subs="+quotes,attributes"]
+----
+- name: _My Playbook_
+  hosts: _localhost_
+  vars_files:
+    - _My_Vault.yml_
+  module_defaults:
+    group/{ansible-module-defaults-group}:
+      password: "{{ _My_Password_ }}"
+  tasks:
+----
+.. In the `tasks:` section, define the tasks that you want your playbook to perform.
+For examples, see xref:example-playbooks-based-on-modules-from-{project-context}-ansible-collections[].
+. Validate the playbook syntax:
++
+[subs="+quotes"]
+----
+$ ansible-playbook --syntax-check --ask-vault-pass ~/_My_Playbook.yml_
```

Comment:
I'd drop the `~/` here (and below), and assume the user placed it somewhere sensible (after all, you didn't tell them to create exactly `~/my_playbook.yml`)

Reply:
Updated.